structs.go

```diff
@@ -946,19 +946,19 @@ type GuildScheduledEventUser struct {
 	Member                *Member `json:"member"`
 }
 
-// A GuildTemplate represents
+// A GuildTemplate represents a replicable template for guild creation
 type GuildTemplate struct {
 	// The unique code for the guild template
 	Code string `json:"code"`
 
 	// The name of the template
-	Name string `json:"name"`
+	Name *string `json:"name,omitempty"`
```

Comment:
While making it a pointer would be good for consistency, it cannot be set to `""` (in contrast to `description`).
So I think just `omitempty` would be enough.
```suggestion
	Name string `json:"name,omitempty"`
```